File: src/main/java/com/polygloat/exceptions/BadRequestException.java
package com.polygloat.exceptions;

import com.polygloat.constants.Message;
import org.springframework.http.HttpStatus;

import java.io.Serializable;
import java.util.List;

public class BadRequestException extends ErrorException {
    public BadRequestException(Message message, List<Serializable> params) {
        super(message, params);
    }

    public BadRequestException(Message message) {
        super(message);
    }

    public HttpStatus getHttpStatus() {
        return HttpStatus.BAD_REQUEST;
    }
}

File: backend/data/src/main/kotlin/io/tolgee/exceptions/BadRequestException.kt
package io.tolgee.exceptions

import org.springframework.http.HttpStatus
import java.io.Serializable

open class BadRequestException : ErrorException {
  constructor(message: io.tolgee.constants.Message, params: List<Serializable>?) : super(message, params)
  constructor(message: io.tolgee.constants.Message) : super(message)
  constructor(code: String, params: List<Serializable>? = null) : super(code, params)

  override val httpStatus: HttpStatus?
    get() = HttpStatus.BAD_REQUEST
}
